pyresample/utils/cf.py

```diff
@@ -0,0 +1,497 @@
+
+    # find the ll: lower-left and ur: upper-right.
+    # x['first'], y['first'] is always the Upper Left corner
+    #   (think of numpy's convention for a 2D image with index 0,0 in top left).
+    ll_x, ll_y = x['first'], y['last']
+    ur_x, ur_y = x['last'], y['first']
+
+    # handle the half-pixel offset between the center of corner cell (what we have in the axis info)
+    #   and the corner of corner cell (what AreaDefinition expects)
+    ll_x -= x['sign'] * 0.5 * x['spacing']
+    ur_x += x['sign'] * 0.5 * x['spacing']
+    ll_y += y['sign'] * 0.5 * y['spacing']
+    ur_y -= y['sign'] * 0.5 * y['spacing']
+
+    # return as tuple
+    ret = (ll_x, ll_y, ur_x, ur_y)
+
+    return ret
+
+
+def _guess_cf_axis_varname(nc_handle, variable, axis, type_of_grid_mapping):
+    """ guess the name of the coordinate variable holding the axis. although Y and X are recommended to
+          be placed last in the list of coordinates, this is not required. """
+
+    ret = None
+
+    if axis not in ('x', 'y'):
+        raise ValueError("axis= parameter must be 'x' or 'y'")
+
+    # the name of y and x are in the dimensions of the variable=
+    try:
+        dims = nc_handle[variable].dims
+    except KeyError:
+        raise ValueError("variable {} not found in file".format(variable))
+
+    for dim in dims:
+        # test if each dim is a valid CF coordinate variable
+        if _is_valid_coordinate_variable(nc_handle, dim, axis, type_of_grid_mapping):
+            ret = dim
+            break
+
+    return ret
+
+
+def _guess_cf_lonlat_varname(nc_handle, variable, lonlat):
+    """ guess the name of the variable holding the latitude (or longitude)
+            corresponding to 'variable' """
+
+    ret = None
+
+    if lonlat not in ('lon', 'lat'):
+        raise ValueError("lonlat= parameter must be 'lon' or 'lat'")
+
+    # lat/lon are either directly a dimension, or a :coordinates.
+
+    # By default (decode_cf=True) xarray puts all dims and :coordinates in .coords
+    #   and remove the :coordinates attribute
+    try:
+        search_list = list(nc_handle[variable].coords)
+    except KeyError:
+        raise ValueError("variable {} not found in file".format(variable))
+
+    # if decode_cf=False was used, the look at the :coordinates attribute
+    if 'coordinates' in nc_handle[variable].attrs.keys():
+        search_list += (nc_handle[variable].attrs['coordinates']).split()
+
+    # go through the list of variables and check if one of them is lat / lon
+    for v in search_list:
+        try:
+            # this allows for both 'latitude' and 'rotated_latitude'...
+            if {'lat': 'latitude', 'lon': 'longitude'}[lonlat] in getattr(nc_handle[v], 'standard_name'):
+                ret = v
+                break
+        except AttributeError:
+            # no 'standard_name'. this is not what we are looking for.
+            pass
+
+    return ret
+
+
+def _load_cf_area_oneVariable(nc_handle, variable, y=None, x=None):
```

Comment:
The function name shouldn't contain capital letters.
Also, the function is very long, and has clearly identified section. I would like to see each section as an independent functions if possible.

Reply:
Renamed and split: https://github.com/TomLav/pyresample/commit/fbc897dfc9806b7a447ae6918af26f2a87d33954